### PR TITLE
Multi-node deployment behind ELB

### LINKED ISSUE #88: Create Production deployment AWS Cloudformation template for TorchServe
For Production deployment of TorchServe, provide AWS Cloudformation templates that support following configurations:

- [x] Simple EC2 deployment
- [ ] Sagemaker deployment with autoscaling
- [ ] K8s/EKS deployment with autoscaling 

Each of the production deployment templates should include configuration with:

- TorchServe inference node(s)
- TorchServe Model management node
- NGINX loadbalancer
- Metrics collection node - Statsd/Prometheus/CloudWatch
- Visualization Dashboard for metrics - Cloudwatch dashboard / Grafana
- Log aggregation on CloudWatch/Fluentd (or similar) 
- Secure deployment option using HTTPS 
- Option to select CPU or GPU deployment for the Inference nodes (this could be a parameter to select AWS instance type for the Inference nodes).

### PR DESCRIPTION
**Pull request by maaquib**
##  Description

Adding cfn template for multi-node ec2 deployment behind an elb

Fixes #88

## Type of change
- [X] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the tests [UT/IT] that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced. 

## Checklist:

- [x] Have you made corresponding changes to the documentation?

## Testing:
Screen Shot 2020-07-22 at 3 52 17 PM
Screen Shot 2020-07-22 at 3 52 54 PM

```console
curl "http://torch-Appli-1A0CZZIDLKO67-848858666.us-east-1.elb.amazonaws.com:8081/models"
{
  "models": [
    {
      "modelName": "squeezenet1_1",
      "modelUrl": "squeezenet1_1.mar"
    }
  ]
}

curl -X POST "http://torch-Appli-1A0CZZIDLKO67-848858666.us-east-1.elb.amazonaws.com:8080/predictions/squeezenet1_1" -T /tmp/kitten.jpg
[
  {
    "tabby": 0.2752000391483307
  },
  {
    "lynx": 0.25468742847442627
  },
  {
    "tiger_cat": 0.24254192411899567
  },
  {
    "Egyptian_cat": 0.22137421369552612
  },
  {
    "cougar": 0.0022544802632182837
  }
]
```

### CODE REVIEW

File: cloudformation/ec2-asg.yaml [hunk 1]
```diff
@@ -0,0 +1,603 @@
...
+            - TorchServeSetup
+            - CloudwatchSetup
+            - CFNSetup
+        TorchServeSetup:
+          files:
+            /etc/torchserve/config.properties:
+               content: !Sub |
+                 inference_address=http://0.0.0.0:8080
+                 management_address=http://0.0.0.0:8081
+                 private_key_file=/etc/torchserve/server.key
```
Comment (dhaniram-kshirsagar) [inline cloudformation/ec2-asg.yaml:435]: if mgmt and inference addresses set are HTTP then do we need these key and cert config?

Comment (maaquib) [reply cloudformation/ec2-asg.yaml:435]: @dhanainme Good catch. Will remove

**Comment by dhanainme** [comment]
Code looks good to me. Please add testing info - screenshot.

**Comment by maaquib** [comment]
> Code looks good to me. Please add testing info - screenshot.

Done!

**Pull request merged**